File: app/src/main/java/com/hjc/base/dialog/PlaneInfoDialog.java
package com.hjc.base.dialog;

import android.app.Dialog;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;
import android.widget.LinearLayout;

import androidx.annotation.Nullable;

import com.blankj.utilcode.util.BarUtils;
import com.hjc.base.R;
import com.hjc.base.databinding.DialogPlaneInfoBinding;
import com.hjc.base.viewmodel.CommonViewModel;
import com.hjc.library_base.dialog.BaseFragmentDialog;

/**
 * @Author: HJC
 * @Date: 2021/9/26 14:03
 * @Description: 飞机信息dialog
 */
public class PlaneInfoDialog extends BaseFragmentDialog<DialogPlaneInfoBinding, CommonViewModel> {

    public static PlaneInfoDialog newInstance() {
        return new PlaneInfoDialog();
    }

    @Override
    public int getStyleRes() {
        return R.style.Base_Dialog_Left;
    }

    @Override
    public int getLayoutId() {
        return R.layout.dialog_plane_info;
    }

    @Override
    public CommonViewModel createViewModel() {
        return null;
    }

    @Override
    protected int getWidth() {
        return WindowManager.LayoutParams.WRAP_CONTENT;
    }

    @Override
    protected int getHeight() {
        return WindowManager.LayoutParams.MATCH_PARENT;
    }

    @Override
    public void initView() {
        super.initView();
        LinearLayout.LayoutParams layoutParams = (LinearLayout.LayoutParams) mBindingView.llLeft.getLayoutParams();
        layoutParams.leftMargin = BarUtils.getStatusBarHeight();
        mBindingView.llLeft.setLayoutParams(layoutParams);
    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {
        Dialog dialog = getDialog();
        if (dialog != null && dialog.getWindow() != null) {
            dialog.getWindow().setDimAmount(0f);
        }
    }

    @Override
    public void addListeners() {

    }

    @Override
    public void onSingleClick(View v) {

    }

}

File: module-other/src/main/java/com/hjc/module_other/dialog/PlaneInfoDialog.kt
package com.hjc.module_other.dialog

import android.content.DialogInterface
import com.hjc.library_base.dialog.BaseFragmentDialog
import com.hjc.library_common.viewmodel.CommonViewModel
import com.hjc.module_other.R
import android.view.WindowManager
import android.widget.LinearLayout
import com.blankj.utilcode.util.BarUtils
import android.os.Bundle
import android.view.View
import com.hjc.library_base.event.EventManager
import com.hjc.library_base.event.MessageEvent
import com.hjc.library_common.global.EventCode
import com.hjc.module_other.databinding.OtherDialogPlaneInfoBinding

/**
 * @Author: HJC
 * @Date: 2021/9/26 14:03
 * @Description: 飞机信息dialog
 */
class PlaneInfoDialog : BaseFragmentDialog<OtherDialogPlaneInfoBinding, CommonViewModel>() {

    companion object {
        fun newInstance(): PlaneInfoDialog {
            return PlaneInfoDialog()
        }
    }

    override fun getStyleRes(): Int {
        return R.style.Base_Dialog_Right
    }

    override fun getLayoutId(): Int {
        return R.layout.other_dialog_plane_info
    }

    override fun createViewModel(): CommonViewModel? {
        return null
    }

    override fun getWidth(): Int {
        return WindowManager.LayoutParams.WRAP_CONTENT
    }

    override fun getHeight(): Int {
        return WindowManager.LayoutParams.MATCH_PARENT
    }

    override fun initView() {
        super.initView()
        val layoutParams = mBindingView.llInfo.layoutParams as LinearLayout.LayoutParams
        layoutParams.rightMargin = BarUtils.getStatusBarHeight()
        mBindingView.llInfo.layoutParams = layoutParams
    }

    override fun initData(savedInstanceState: Bundle?) {
        dialog?.window?.setDimAmount(0f)
    }

    override fun addListeners() {
        mBindingView.onClickListener = this
    }

    override fun onSingleClick(v: View?) {
        when (v?.id) {
            R.id.ll_left -> {
                dismiss()
            }

            else -> {
            }
        }
    }

    override fun onDismiss(dialog: DialogInterface) {
        super.onDismiss(dialog)
        EventManager.sendEvent(MessageEvent(EventCode.HIDE_PLANE_VIEW, null))
    }

}